lib/core.ml

```diff
@@ -53,11 +51,15 @@ type state =
     desc : Automata.State.t
     (* Description of this state of the automata *) }
 
+(* Transition table, indexed by color. The state information is stored
+   at position 0. *)
+type table = Table of table array [@@unboxed]
```

Comment:
I think you should isolate `table` and `state` in a module, rename them (old `state` is now pretty much `info`, and new `table` should be named `state`.

Also, I would suggest adding a comment that explains you want a contiguous memory range representing `state_info * table`, but OCaml can't do that for you without adding an indirection, so you do it manually.

Is there any reason not to inline also `idx`? It is also accessed in the critical path.